src/compiler/checker.ts

```diff
@@ -24987,14 +24987,14 @@ namespace ts {
         }
 
         // No signature was applicable. We have already reported the errors for the invalid signature.
-        // If this is a type resolution session, e.g. Language Service, try to get better information than anySignature.
         function getCandidateForOverloadFailure(
             node: CallLikeExpression,
             candidates: Signature[],
             args: readonly Expression[],
             hasCandidatesOutArray: boolean,
         ): Signature {
             Debug.assert(candidates.length > 0); // Else should not have called this.
+            resolveUntypedCall(node);

```

Comment:
@sandersn do you want to consider deferring this to the end of somehow (since it just needs to happen once for errors to be reported and usage to be painted, rather than on every attempted resolution)? Like how we have `checkNodeDeferred`? This does fix parameter types, and looking at the baselines, the parameter types we get from deferring this may be nicer that what's here.

Like [this change](https://github.com/weswigham/TypeScript/commit/465c0c644ded48a0c21279cc7715d83794f43cba) is what I'm proposing - it looks like it minimizes the baseline diff a lot.

Reply:
Seems reasonable. I'll give it a try.